### ISSUE TITLE
ALL FragmentLossType apparently NaN

### ISSUE THREAD

**Issue by JM-Bader** [Nesvilab/FragPipe#677]
Hi, i would like to use MSFragger to generate a library from DDA PASEF data for a subsequent diaPASEF search with DIA-NN.
Overall, this works. However, when looking at the library outputted by fragpipe running MSFragger, it is striking that all entries in FragmentLossType are NaN.  This came up as DIA-NN raised an alarm and I was first in touch with Demichev. He said that DIA-NN treats NaN as no loss by default, so it is possible to run though. However, that clearly isn't ideal as some of these entries very likely have a neutral loss. 

My Library: 
![image](https://user-images.githubusercontent.com/98888664/168096436-5f77a1bd-f082-4199-8b97-deb073d02017.png)

MSFragger settings in fragpipe: 
![image](https://user-images.githubusercontent.com/98888664/168096597-a5127d05-bed1-4fc0-8392-b305d06f1616.png)
![image](https://user-images.githubusercontent.com/98888664/168096851-39af246b-f67c-406a-bc07-31802763027a.png)

general fragpipe version etc.:
![image](https://user-images.githubusercontent.com/98888664/168096968-07eb7915-15f1-45a8-95f5-f2b6703ee9f9.png)

Q1: What causes the issue?
Q2: How to fix it? (ie. get the library with relevant FragmentLossType information)

**Comment by JM-Bader** [comment]
How does setting 'Deneutralloss' to NO impact this ? (e.g. in the new version 18.0)?
![image](https://user-images.githubusercontent.com/98888664/171664647-16eb5290-bdce-4bd6-ab99-1c5b5dc11190.png)
Is neutral loss information available there now?

**Comment by fcyu** [comment]
Setting it to "No" might result in worse identification result. It is used in MSFragger database searching. Not for spectral library building.

Best,

Fengchao

**Comment by JM-Bader** [comment]
Thanks for the swift reply. 
So the new version 18.0 that just came out does not yet support neutral loss specification for libraries (independent from the 'Deneutralloss' button), correct?

**Comment by fcyu** [comment]
Correct. Writing neutral loss info to the spectral library is not yet supported. We will implement it in the near future.

Best,

Fengchao

**Comment by anesvi** [comment]
JM-Bader: If you contact us directly (e.g. email me), I want to understand your interest in the neutral losses. Just regular -18/-17 are not likely to be that useful, I think. We will add them, though.

***@***.***>Guo Ci: can you add an option to write neutral losses in the library. It is already in EasyPQP we just need to propagate that option to FragPipe. And we need to check the format, if the new column as written by EasyPQP is compatible with 1.8.1. If not, we will email Vadim (or will need to change to DAI-NN format)

Alexey

**Comment by fcyu** [comment]
The format/string that can be recognized by DIA-NN is as following.

Best,

Fengchao

> Following is the code from Vadim ([vdemichev/DiaNN#377 (reply in thread)](https://github.com/vdemichev/DiaNN/discussions/377#discussioncomment-2738553)):
> 
> ```
> if (wl == "noloss") p.loss = loss_none;
> 									else if (wl == "H2O") p.loss = loss_H2O;
> 									else if (wl == "NH3") p.loss = loss_NH3;
> 									else if (wl == "CO") p.loss = loss_CO;
> 									else if (wl == "H3PO4") p.loss = loss_H3PO4;
> 									else if (wl == "H4COS") p.loss = loss_H4COS;
> 									else p.loss = loss_other;
> ```
> 
> Hi @guoci , we need to add checkboxes in the `spectral library generation` tab to support neutral loss. I can change the FragPipe part, but need your help to change the Python script and EasyPQP.
> 
> Best,
> 
> Fengchao

**Comment by fcyu** [comment]
Hi @JM-Bader ,

We have added the neutral loss support to the spectral library building of FragPipe. Please contact yufe AT umich.edu if you want to test the pre-release version.

Best,

Fengchao
